### PR TITLE
feat:adds asmcli to the krm dev tools image

### PR DESCRIPTION
**Pull request by g-awmalik**
This is required for installing asm during int. testing for policy controller.

### CODE REVIEW

File: infra/build/developer-tools-krm/Dockerfile [hunk 1]
```diff
@@ -20,6 +20,13 @@ RUN apk update && apk add --no-cache openrc docker-cli docker screen nodejs-curr
 # Additional go tooling
 RUN GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.40.1
 
+# Required to download and install asmcli
+ARG ASMCLI_VERSION
+ENV ASMCLI_VERSION ${ASMCLI_VERSION}
+
+ADD ./build/install_asmcli.sh /build/
+RUN /build/install_asmcli.sh ${ASMCLI_VERSION}
+
```
Comment (bharathkkb) [inline infra/build/developer-tools-krm/Dockerfile:29]: We can remove /build since we wont need it after install

File: infra/build/Makefile [hunk 2]
```diff
@@ -29,6 +29,7 @@ TERRAGRUNT_VERSION := 0.25.3
 KUSTOMIZE_VERSION := 3.6.0
 KPT_VERSION := 1.0.0-beta.13
 CFT_CLI_VERSION := 0.4.2
+ASMCLI_VERSION := 1.12
 
 DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 1.4.1
```
Comment (bharathkkb) [inline infra/build/Makefile:34]:
```suggestion
DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 1.4.2
```

**Review by bharathkkb — APPROVED** [review]

File: infra/build/developer-tools-krm/Dockerfile (hunk 1 above)
Comment (g-awmalik) [reply infra/build/developer-tools-krm/Dockerfile:29]: is this what you mean?
```
ADD ./build/install_asmcli.sh .
RUN install_asmcli.sh ${ASMCLI_VERSION}
```
Having the install script at root doesn't seem right. Just to be clear this isn't install ASM yet. This is only setting up the utility to install ASM later.

**Comment by g-awmalik** [comment]
@bharathkkb - i updated the version for kubectl as well in the base image.

File: infra/build/developer-tools-krm/Dockerfile (hunk 1 above)
Comment (bharathkkb) [reply infra/build/developer-tools-krm/Dockerfile:29]: No, I meant 

```
ADD ./build/install_asmcli.sh /build/
RUN /build/install_asmcli.sh ${ASMCLI_VERSION}
RUN rm -rf /build
```
This way `/build/install_asmcli.sh` is not persisted as we dont need it after the CLI is installed.

> Having the install script at root doesn't seem right. 

Yeah we can continue to keep the script in /build which we can then clean up.

> Just to be clear this isn't installing ASM yet. This is only setting up the utility to install ASM later.

Yup, understood that this is just installing the cli and not installing ASM.

Comment (g-awmalik) [reply infra/build/developer-tools-krm/Dockerfile:29]: Got it. I'll remove that.

**Comment by g-awmalik** [comment]
@bharathkkb - can you take another look? thnx

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**